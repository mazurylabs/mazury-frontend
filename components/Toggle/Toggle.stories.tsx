import { ComponentMeta, ComponentStory } from '@storybook/react';
import { useState } from 'react';
import { Toggle } from './Toggle';

export default {
  title: 'Components/Toggle',
  component: Toggle,
  argTypes: {
    isToggled: {
      type: 'boolean',
    },
  },
} as ComponentMeta<typeof Toggle>;

export const Default = () => {
  const [active, setActive] = useState(false);

  return <Toggle isToggled={active} onToggle={setActive} />;
};
