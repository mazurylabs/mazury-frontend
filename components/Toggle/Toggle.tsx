import { FC, useRef } from 'react';
import SVG from 'react-inlinesvg';

interface ToggleProps {
  isToggled: boolean;
  onToggle: (active: boolean) => void;
  className?: string;
  id?: string;
}

export const Toggle: FC<ToggleProps> = ({
  isToggled,
  onToggle,
  className,
  id,
}) => {
  const hiddenCheckboxRef = useRef<HTMLInputElement>(null);

  const onClick = () => {
    hiddenCheckboxRef?.current?.click();
  };

  return (
    <div className={`${className}`} data-testid="toggle-outer-container">
      <button type="button" onClick={onClick} className="hover:cursor-pointer">
        <SVG
          src={`/icons/toggle-${isToggled ? 'on' : 'off'}.svg`}
          width="40px"
          height="24px"
        />
      </button>
      <input
        data-testid="Hidden Checkbox"
        ref={hiddenCheckboxRef}
        type="checkbox"
        hidden
        checked={isToggled}
        onChange={(e) => onToggle(e.target.checked)}
        id={id}
      />
    </div>
  );
};
