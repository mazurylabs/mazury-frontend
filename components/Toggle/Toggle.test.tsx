import { Toggle } from './Toggle';
import { useState } from 'react';
import { fireEvent, render, screen } from '@testing-library/react';

const WrappedToggle = () => {
  const [isToggled, setToggled] = useState(false);

  return (
    <Toggle
      isToggled={isToggled}
      onToggle={() => setToggled((isToggled) => !isToggled)}
    />
  );
};

test('Toggle', () => {
  render(<WrappedToggle />);
  const outerContainer = screen.getByTestId('toggle-outer-container');
  expect(outerContainer).toBeInTheDocument();
  expect(outerContainer).toMatchSnapshot();
  const toggle = screen.getByRole('button', { name: 'Toggle' });
  const hiddenCheckbox = screen.getByTestId(
    'Hidden Checkbox'
  ) as HTMLInputElement;
  expect(hiddenCheckbox).toBeInTheDocument();
  expect(hiddenCheckbox).not.toBeChecked();
  expect(toggle).toBeInTheDocument();
  fireEvent.click(toggle);
  expect(hiddenCheckbox).toBeChecked();
  expect(toggle).toMatchSnapshot();
});
