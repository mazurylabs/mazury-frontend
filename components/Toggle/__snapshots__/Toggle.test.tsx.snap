// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle 1`] = `
<div
  class="undefined"
  data-testid="toggle-outer-container"
>
  <button
    class="hover:cursor-pointer"
    type="button"
  />
  <input
    data-testid="Hidden Checkbox"
    hidden=""
    type="checkbox"
  />
</div>
`;
